### ISSUE TITLE
Null returned from message creator is not handled

### ISSUE THREAD

**Issue by cpx86** [danielwertheim/routemeister#8]
If a message creator returns a null value, Routemeister will still invoke the handler method but with a null reference for the handler instance. This is technically allowed by reflection, but not correct and leads to some unexpected behaviors. A realistic case where this might happen is if you plugin a container that returns default values instead of throwing exceptions when it can't resolve a type.

I've gone through the code and adding a simple null-check seems to be trivial and should hardly have any performance impact. Would you be interested in a PR?

Here's a simple repro:
```csharp
void Main()
{
    var factory = new MessageRouteFactory();
    var routes = new MessageRoutes
    {
        factory.Create(new[] {GetType().GetTypeInfo().Assembly}, typeof (IHandle<>))
    };
    var router = new SequentialAsyncRouter((t, e) => null, routes);

    router.RouteAsync("hello");
}

interface IHandle<T>
{
    Task HandleAsync(T message);
}

public class StringHandler : IHandle<string>
{
    public Task HandleAsync(string message)
    {
        Console.WriteLine(this.ToString()); // this is null
        return Task.CompletedTask;
    }
}
```

**Comment by danielwertheim** [comment]
That sounds fair. I'll gladly accept "something" preventing it from
accepting null handlers being resolved.

**Comment by cpx86** [comment]
I was thinking to throw `InvalidOperationException` if the handler is null. Sound good?

One question about the publish methods - should it resolve all handlers first and validate them before attempting to execute them? Also, slightly off topic, but currently if one handler throws an exception then subsequent handlers will not be executed. Is that by design?

**Comment by danielwertheim** [comment]
Regarding the subsequent handlers, yes, the router and dispatchers are pretty dumb by intention. The idea would be to either create a more advanced using the Routes and contribute if it makes sense. And just to be an in-memory routing/dispatching solution.

Regarding `InvalidOperationException`, sounds fair.

The pre-resolving sounds nice, but not sure if they should be re-used. Thinking the strategy of lifetime of a handler is up to the creator, but I guess it's fair to think "per-message".

Now to a bit of.... The same goes for Dispatcher(s) which doesn't use the router under the hood. This should probably be taken care of in a separate PR/Issue.

**Comment by cpx86** [comment]
No, it's probably better not to re-use them. Was just thinking that it could make sense to first ensure all handlers can actually be resolved before executing any of them.

I'll prepare a PR for this issue now then.
